python/lsst/ts/salobj/sal_info.py

```diff
@@ -781,7 +785,8 @@ def _blocking_read(
         conditions = self._waitset.wait(self._wait_timeout)
         if not self.isopen:
             # shutting down; clean everything up
-            return (None, [])
+            return None
+        ret = []
```

Comment:
This needs a type annotation to make mypy happy

Reply:
Please make this change (you marked this resolved without changing it).

Reply:
Also please put:
```
if not self.isopen:
    return []
```
just after `conditions = ...`, or to be even more cautious, just after `for condition in conditions:`, to avoid unexpected warning messages at shutdown. self.isopen is set false earlier than ReadTopic.isopen, with the event loop running in between, and once self.isopen is false there is no point to continuing.  We can live without it, but I'd rather play it safe for fear of introducing unexpected warnings at shutdown.